src/main/java/com/autotune/common/target/kubernetes/service/impl/KubernetesServicesImpl.java

```diff
@@ -0,0 +1,492 @@
+     * Replace with new deployment.
+     *
+     * @param namespace
+     * @param deploymentName
+     * @param newDeployment
+     * @return
+     */
+    public boolean replaceDeployment(String namespace, String deploymentName, Deployment newDeployment) {
+        boolean deployed = false;
+        try {
+            kubernetesClient
+                    .apps()
+                    .deployments()
+                    .inNamespace(namespace)
+                    .withName(deploymentName)
+                    .createOrReplace(newDeployment);
+            deployed = true;
+        } catch (Exception e) {
+            new TargetHandlerException(e, "replaceDeployment failed!");
+        }
+        return deployed;
+    }
+
+    /**
+     * Restart deployment. Used by EM to restart deployment during warmup/measurements cycles.
+     *
+     * @param namespace
+     * @param deploymentName
+     * @return
+     */
+    @Override
+    public boolean restartDeployment(String namespace, String deploymentName) {
+        boolean restarted = false;
+        try {
+            kubernetesClient
+                    .apps()
+                    .deployments()
+                    .inNamespace(namespace)
+                    .withName(deploymentName)
+                    .rolling()
+                    .restart();
+            restarted = true;
+        } catch (Exception e) {
+            new TargetHandlerException(e, "restartDeployment failed!");
+        }
+        return restarted;
+    }
+
+    /**
+     * start Deploying using following params. Used by EM to apply config from trials
+     *
+     * @param namespace
+     * @param deploymentName
+     * @param containerConfigData
+     * @return
+     */
+    @Override
+    public boolean startDeploying(String namespace, String deploymentName, ContainerConfigData containerConfigData) {
+        boolean deployed = false;
+        try {
+            Deployment existingDeployment = getDeploymentBy(namespace, deploymentName);
+            if (existingDeployment != null) {
+                Deployment newDeployment = getAmendedDeployment(namespace, deploymentName, existingDeployment, containerConfigData);
+                if (newDeployment != null) {
+                    replaceDeployment(namespace, deploymentName, newDeployment);
+                    deployed = true;
+                }
+            } else {
+                throw new Exception("Deployment does not exist.");
+            }
+        } catch (Exception e) {
+            new TargetHandlerException(e, "deployDeployment failed!");
+        }
+        return deployed;
+    }
+
+    /**
+     * Get modified deployment as per configs.
+     *
+     * @param namespace
+     * @param deploymentName
+     * @param existingDeployment
+     * @param containerConfigData
+     * @return
+     */
+    public Deployment getAmendedDeployment(String namespace, String deploymentName, Deployment existingDeployment, ContainerConfigData containerConfigData) {
```

Comment:
This should be called `amendDeployment` as `get` gives an impression that it is read-only, whereas this is where we are making a change

Reply:
done